packages/eslint-plugin/src/rules/consistent-indexed-object-style.ts

```diff
@@ -95,12 +95,46 @@ export default createRule<Options, MessageIds>({
         return;
       }
 
+      const scope = context.getScope();
+      if (
+        scope.block.type == AST_NODE_TYPES.Program &&
+        scope.block.body[0].type == AST_NODE_TYPES.TSTypeAliasDeclaration
+      ) {
+        const body = scope.block.body[0];
+        const name = body?.id.name;
+        const memberTypes = member.typeAnnotation?.typeAnnotation;
+        if (memberTypes) {
+          if (
+            memberTypes.type == AST_NODE_TYPES.TSTypeReference &&
+            memberTypes.typeName.type == AST_NODE_TYPES.Identifier &&
+            memberTypes.typeName.name == name
+          ) {
+            return;
+          } else if (memberTypes.type == AST_NODE_TYPES.TSUnionType) {
+            const membersArray = memberTypes.types;
+            let flag = false;
+            membersArray.forEach((m: TSESTree.Node) => {
+              if (
+                m.type == AST_NODE_TYPES.TSTypeReference &&
+                m.typeName.type == AST_NODE_TYPES.Identifier &&
+                m.typeName.name == name
+              ) {
+                flag = true;
+              }
+            });
+            if (flag) {
+              return;
+            }
```

Comment:
You can use a `for-of` loop instead of `.forEach` and that allows you to return directly instead of using the `flag`:

```suggestion
            const membersArray: TSESTree.Node[] = memberTypes.types;
            for (const m of membersArray) {
              if (
                m.type == AST_NODE_TYPES.TSTypeReference &&
                m.typeName.type == AST_NODE_TYPES.Identifier &&
                m.typeName.name == name
              ) {
                return;
              }
            }
```